src/android/CleverTapPlugin.java

```diff
@@ -202,12 +201,8 @@ public boolean execute(String action, JSONArray args, final CallbackContext call
         boolean haveError = false;
         String errorMsg = "unhandled CleverTapPlugin action";
 
-        PluginResult result = null;
-
         if (!checkCleverTapInitialized()) {
-            result = new PluginResult(PluginResult.Status.ERROR, "CleverTap API not initialized");
-            result.setKeepCallback(true);
-            callbackContext.sendPluginResult(result);
+            sendPluginResult(callbackContext, Status.ERROR, "CleverTap API not initialized");
             return true;
         }
 
```

Comment:
This if-else tree is just awful. Can we try to simplify it at least a bit:
* The first `else` is not needed since the first if returns. This is also true for many `else if`s below, but they seem to at least compare on the same thing, so have common logical path.
* We can have all actions comparison logic in a switch statement
* You could consider adding the actions to an Enum, creating the enum value from the action string and then having the switch by the enum value. This would be useful if those actions are also used elsewhere.

Reply:
Can do this in a separate PR since it will need a lot of formatting and basically revamping the entire file

Reply:
I would prefer the change inside the same PR since there are already a lot of changes to the file. If it is in its own commit I'll be able to track the changes.

Reply:
Adding this to tech debt due to other higher priority tasks

https://wizrocket.atlassian.net/browse/SDK-4191